exercises/binary-search/binary_search_test.go

```diff
@@ -1,193 +1,52 @@
-// Go has binary search in the standard library.  Let's reimplement
-// sort.SearchInts with the same API as documented in the standard library
-// at http://golang.org/pkg/sort/#Search.  Note that there are some differences
-// with the exercise README.
-//
-// * If there are duplicate values of the key you are searching for, you can't
-// just stop at the first one you find; you must find the first occurrence in
-// the slice.
-//
-// * There is no special "not found" indication.  If the search key is not
-// present, SearchInts returns the index of the first value greater than the
-// search key.  If the key is greater than all values in the slice, SearchInts
-// returns the length of the slice.
-//
-// * You can assume the slice is sorted in increasing order.  There is no need
-// to check that it is sorted.  (That would wreck the performance.)
-//
-// Try it on your own without peeking at the standard library code.
-
 package binarysearch
 
 import (
+	"fmt"
 	"math/rand"
-	"sort"
 	"testing"
 )
 
-var testData = []struct {
-	ref   string
-	slice []int
-	key   int
-	x     int // expected result
-}{
-	{"6 found at index 3",
-		[]int{1, 3, 4, 6, 8, 9, 11},
-		6, 3},
-	{"9 found at index 5",
-		[]int{1, 3, 4, 6, 8, 9, 11},
-		9, 5},
-	{"3 found at index 1",
-		[]int{1, 3, 5, 8, 13, 21, 34, 55, 89, 144},
-		3, 1},
-	{"55 found at index 7",
-		[]int{1, 3, 5, 8, 13, 21, 34, 55, 89, 144},
-		55, 7},
-	{"21 found at index 5",
-		[]int{1, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377},
-		21, 5},
-	{"34 found at index 6",
-		[]int{1, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377},
-		34, 6},
-
-	{"1 found at beginning of slice",
-		[]int{1, 3, 6},
-		1, 0},
-	{"6 found at end of slice",
-		[]int{1, 3, 6},
-		6, 2},
-	{"2 > 1 at index 0, < 3 at index 1",
-		[]int{1, 3, 6},
-		2, 1},
-	{"2 < all values",
-		[]int{11, 13, 16},
-		2, 0},
-	{"21 > all 3 values",
-		[]int{11, 13, 16},
-		21, 3},
-	{"1 found at beginning of slice",
-		[]int{1, 1, 1, 1, 1, 3, 6}, // duplicates
-		1, 0},
-	{"3 found at index 1",
-		[]int{1, 3, 3, 3, 3, 3, 6},
-		3, 1},
-	{"6 found at index 4",
-		[]int{1, 3, 3, 3, 6, 6, 6},
-		6, 4},
-	{"-2 > -3 at index 1, < -1 at index 2",
-		[]int{-6, -3, -1}, // negatives
-		-2, 2},
-	{"0 > -7 at index 4, < 1 at index 5",
-		[]int{-19, -17, -15, -12, -7, 1, 14, 35, 69, 124},
-		0, 5},
-	{"slice has no values",
-		nil, 0, 0},
-}
-
 func TestSearchInts(t *testing.T) {
-	for _, test := range testData {
-		if !sort.IntsAreSorted(test.slice) {
-			t.Skip("Invalid test data")
-		}
+	for _, test := range testCases {
 		if x := SearchInts(test.slice, test.key); x != test.x {
-			t.Fatalf("SearchInts(%v, %d) = %d, want %d",
+			t.Fatalf("FAIL: SearchInts(%#v, %d) = %d, want %d",
 				test.slice, test.key, x, test.x)
```

Comment:
Since you have a  test.description, perhaps include it in first line of output, with SearchInts ... on subsequent line?

Reply:
Thanks, updated so.